code/modules/mob/living/say.dm

```diff
@@ -350,6 +354,23 @@ var/list/department_radio_keys = list(
 						handle_render(M, message,src)
 				if((M in dead_mob_list) && !istype(M, /mob/new_player))
 					handle_render(M, message,src)
+		if(MODE_BORER)
+			//this is sent to and usable by borers and mobs controlled by borers
+			var/mob/living/simple_animal/borer/head = src.has_brain_worms(LIMB_HEAD)
+			if(isborer(src) || head && head.controlling)
+				var/mob/living/simple_animal/borer/B = head && head.controlling ? head : src
+				var/message = text("<span class='cortical'>Cortical link, <b>[]</b>: []</span>",B.truename, html_encode(speech.message))
+				var/turf/T = get_turf(src)
+				log_say("[key_name(src)] (@[T.x],[T.y],[T.z]) Borer chat: [html_encode(speech.message)]")
+				
+				for(var/mob/M in mob_list)
```

Comment:
I think this whole section can be replaced by `for (var/living/simple_animal/borer/B in mob_list)` and another loop for obserbver ; it would be cleaner
Unless I'm misunderstanding what you're doing